pennylane/transforms/decompositions/clifford_t/clifford_t_transform.py

```diff
@@ -0,0 +1,495 @@
+        )
+
+    return check_clifford_op(op, use_decomposition=use_decomposition)
+
+
+def _simplify_param(theta, gate):
+    r"""Check if the parameter allows simplification for the rotation gate.
+
+    For the cases where theta is an integer multiple of π: (a) returns a global phase
+    when even, and (b) returns combination of provided gate with global phase when odd.
+    In rest of the other cases it would return None.
+    """
+    if qml.math.is_abstract(theta):  # pragma: no cover
+        return None
+
+    if qml.math.allclose(theta, 0.0, atol=1e-6):
+        return [qml.GlobalPhase(0.0)]
+
+    rem_, mod_ = qml.math.divide(theta, math.pi), qml.math.mod(theta, math.pi)
+    if qml.math.allclose(mod_, 0.0, atol=1e-6):
+        ops = [qml.GlobalPhase(theta / 2)]
+        if qml.math.allclose(qml.math.mod(rem_, 2), 1.0, atol=1e-6):
+            ops.append(gate)
+        return ops
+
+    return None
+
+
+# pylint: disable= too-many-branches,
+def _rot_decompose(op):
+    r"""Decomposes a rotation operation: :class:`~.Rot`, :class:`~.RX`, :class:`~.RY`, :class:`~.RZ`,
+    :class:`~.PhaseShift` into a basis composed of :class:`~.RZ`, :class:`~.S`, and :class:`~.Hadamard`.
+    """
+    d_ops = []
+    # Extend for Rot operation with Rz.Ry.Rz decompositions
+    if isinstance(op, qml.Rot):
+        (phi, theta, omega), wires = op.parameters, op.wires
+        for dec in [qml.RZ(phi, wires), qml.RY(theta, wires), qml.RZ(omega, wires)]:
+            d_ops.extend(_rot_decompose(dec))
+        return d_ops
+
+    (theta,), wires = op.data, op.wires
+    if isinstance(op, qml.ops.Adjoint):  # pylint: disable=no-member
+        ops_ = _rot_decompose(op.base.adjoint())
+    elif isinstance(op, qml.RX):
+        ops_ = _simplify_param(theta, qml.PauliX(wires=wires))
+        if ops_ is None:  # Use Rx = H @ Rz @ H
+            ops_ = [qml.Hadamard(wires), qml.RZ(theta, wires), qml.Hadamard(wires)]
+    elif isinstance(op, qml.RY):
+        ops_ = _simplify_param(theta, qml.PauliY(wires=wires))
+        if ops_ is None:  # Use Ry = S @ H @ Rz @ H @ S.adjoint()
+            ops_ = [
+                qml.S(wires),
+                qml.Hadamard(wires),
+                qml.RZ(theta, wires),
+                qml.Hadamard(wires),
+                qml.adjoint(qml.S(wires)),
+            ][::-1]
+    elif isinstance(op, qml.RZ):
+        ops_ = _simplify_param(theta, qml.PauliZ(wires=wires))
+        if ops_ is None:
+            ops_ = [qml.RZ(theta, wires)]
+    elif isinstance(op, qml.PhaseShift):
+        ops_ = _simplify_param(theta, qml.PauliZ(wires=wires))
+        if ops_ is None:
+            ops_ = [qml.RZ(theta, wires=wires), qml.GlobalPhase(theta / 2)]
+        else:
+            ops_.append(qml.GlobalPhase(-theta / 2))
+
+    else:
+        raise ValueError(
+            f"Operation {op} is not a supported Pauli rotation: qml.RX, qml.RY, qml.RZ, qml.Rot and qml.PhaseShift."
+        )
```

Comment:
From the perspective of the transform, this comes at the end because by design, I already ensure that I will always call this function with the valid set of gates only. So, putting it in front would only add an extra condition that would fail every time. From the perspective of completeness, this exists for a user/dev who tries to use this independent of the transform.